synapseclient/models/agent.py

```diff
@@ -0,0 +1,447 @@
+        modified_on: The date the agent session was last modified.
+        agent_registration_id: The registration ID of the agent that will be used for this session.
+        etag: The etag of the agent session.
+    """
+
+    id: Optional[str] = None
+    """The unique ID of the agent session. Can only be used by the user that created it."""
+
+    access_level: Optional[
+        AgentSessionAccessLevel
+    ] = AgentSessionAccessLevel.PUBLICLY_ACCESSIBLE
+    """The access level of the agent session.
+        One of PUBLICLY_ACCESSIBLE, READ_YOUR_PRIVATE_DATA, or WRITE_YOUR_PRIVATE_DATA.
+        Defaults to PUBLICLY_ACCESSIBLE.
+    """
+
+    started_on: Optional[datetime] = None
+    """The date the agent session was started."""
+
+    started_by: Optional[int] = None
+    """The ID of the user who started the agent session."""
+
+    modified_on: Optional[datetime] = None
+    """The date the agent session was last modified."""
+
+    agent_registration_id: Optional[int] = None
+    """The registration ID of the agent that will be used for this session."""
+
+    etag: Optional[str] = None
+    """The etag of the agent session."""
+
+    chat_history: List[AgentPrompt] = field(default_factory=list)
+    """A list of AgentPrompt objects."""
+
+    def fill_from_dict(self, synapse_agent_session: Dict[str, str]) -> "AgentSession":
+        """
+        Converts a response from the REST API into this dataclass.
+
+        Arguments:
+            synapse_agent_session: The response from the REST API.
+
+        Returns:
+            The AgentSession object.
+        """
+        self.id = synapse_agent_session.get("sessionId", None)
+        self.access_level = synapse_agent_session.get("agentAccessLevel", None)
+        self.started_on = synapse_agent_session.get("startedOn", None)
+        self.started_by = synapse_agent_session.get("startedBy", None)
+        self.modified_on = synapse_agent_session.get("modifiedOn", None)
+        self.agent_registration_id = synapse_agent_session.get(
+            "agentRegistrationId", None
+        )
+        self.etag = synapse_agent_session.get("etag", None)
+        return self
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Start_Session: {self.id}"
+    )
+    async def start_async(
+        self, *, synapse_client: Optional[Synapse] = None
+    ) -> "AgentSession":
+        """Starts an agent session.
+
+        Arguments:
+            synapse_client: The Synapse client to use for the request. If None, the default client will be used.
+
+        Returns:
+            The new AgentSession object.
+        """
+        syn = Synapse.get_client(synapse_client=synapse_client)
+        session_response = await start_session(
+            access_level=self.access_level.value,
+            agent_registration_id=self.agent_registration_id,
+            synapse_client=syn,
+        )
+        return self.fill_from_dict(session_response)
+
+    @otel_trace_method(
```

Comment:
A bit of a tangent, but could we leverage signoz cloud during python client development?

Reply:
Yes